rclpy/rclpy/node.py

```diff
@@ -264,49 +258,28 @@ def destroy_node(self):
         if self.handle is None:
             return ret
 
-        # ensure that the passed node contains a valid capsule
-        class_ = self.handle.__class__
-        if not class_ or class_.__name__ != 'PyCapsule':
-            raise ValueError('The node handle must be a PyCapsule')
+        for sub in self.subscriptions:
+            _rclpy.rclpy_destroy_node_entity(sub.subscription_handle, self.handle)
+        for pub in self.publishers:
+            _rclpy.rclpy_destroy_node_entity(pub.publisher_handle, self.handle)
+        for cli in self.clients:
+            _rclpy.rclpy_destroy_node_entity(cli.client_handle, self.handle)
+        for srv in self.services:
+            _rclpy.rclpy_destroy_node_entity(srv.service_handle, self.handle)
+        for tmr in self.timers:
+            _rclpy.rclpy_destroy_entity(tmr.timer_handle)
+        for gc in self.guards:
+            _rclpy.rclpy_destroy_entity(gc.guard_handle)
 
-        for sub in list(self.subscriptions):
-            destroyed = _rclpy.rclpy_destroy_node_entity(
-                'subscription', sub.subscription_handle, self.handle)
-            if destroyed:
-                self.subscriptions.remove(sub)
-            ret &= destroyed
-        for pub in list(self.publishers):
-            destroyed = _rclpy.rclpy_destroy_node_entity(
-                'publisher', pub.publisher_handle, self.handle)
-            if destroyed:
-                self.publishers.remove(pub)
-            ret &= destroyed
-        for cli in list(self.clients):
-            destroyed = _rclpy.rclpy_destroy_node_entity(
-                'client', cli.client_handle, self.handle)
-            if destroyed:
-                self.clients.remove(cli)
-            ret &= destroyed
-        for srv in list(self.services):
-            destroyed = _rclpy.rclpy_destroy_node_entity(
-                'service', srv.service_handle, self.handle)
-            if destroyed:
-                self.services.remove(srv)
-            ret &= destroyed
-        for tmr in list(self.timers):
-            destroyed = _rclpy.rclpy_destroy_entity('timer', tmr.timer_handle)
-            if destroyed:
-                self.timers.remove(tmr)
-            ret &= destroyed
-        for gc in list(self.guards):
-            destroyed = _rclpy.rclpy_destroy_entity('guard_condition', gc.guard_handle)
-            if destroyed:
-                self.guards.remove(gc)
-            ret &= destroyed
-        destroyed = _rclpy.rclpy_destroy_entity('node', self.handle)
-        if destroyed:
-            self._handle = None
-        ret &= destroyed
+        self.timers = []
+        self.subscriptions = []
+        self.services = []
+        self.clients = []
+        self.publishers = []
+        self.guards = []
```

Comment:
Nitpick: maybe use the same order as above for the loops? That allows easier comparison that the same variables are used in both.

Reply:
pub, sub, cli, srv, tmr, gc 1561483d1d16434c7c595a3bd8d026df7da2d480